kotlin/reflect/src/main/resources/META-INF/proguard/moshi-kotlin.pro

```diff
@@ -1,5 +1,6 @@
 -keep class kotlin.reflect.jvm.internal.impl.builtins.BuiltInsLoaderImpl
 
+-keep class kotlin.Metadata { *; }
 -keepclassmembers class kotlin.Metadata {
```

Comment:
I'm not as versed in proguard rules as @JakeWharton, but I think the added line effectively invalidates this one by keeping everything in the Metadata class, which is more broad than Moshi needs.

Reply:
That is right

Reply:
Ups... True. To optimize it, I suggest changing original keep option for `kotlin.Metadata`, to:
```
-keep class kotlin.Metadata {
    public <methods>;
}
```
Because if I don't preserve class, I get exception:
```
java.lang.IllegalArgumentException: Cannot serialize Kotlin type my.package.CensoredClass$CensoredSubclass. 
  Reflective serialization of Kotlin classes without using kotlin-reflect has undefined and unexpected behavior. 
  Please use KotlinJsonAdapter from the moshi-kotlin artifact or use code gen from the moshi-kotlin-codegen artifact.
```

Only way that I can explain the error being fixed by this new suggestion, is by the sentence of the [proguard manual](https://www.guardsquare.com/en/products/proguard/manual/usage#keepoptions) manual for `keepclassmembers`:

> Specifies class members to be preserved, if their classes are preserved as well. 

Which I would transfer to: `keepclassmembers`  has no effect if you don't keep the class. 

I might be missing something, because the more I dig into proguard, the less I'm confident with it. Anyway,
just wanted to share something that worked for me.